src/main/java/Parser.java

```diff
@@ -0,0 +1,89 @@
+public class Parser {
+    private TaskList list;
+
+    public Parser(TaskList list) {
+        this.list = list;
+    }
+
+    /**
+     *  Handles the logic flow direction from user input.
+     *  1. When a '/' is present, there are 3 arguments to the input.
+     *  The first loop section handles by splitting them into command, task name and deadline,
+     *  and passes them down to handleThreeInputs(command, name, time, d);
+     *  2. Else, it is a double argument input, split by first empty space " ",
+     *  with split[1] being task name if command is todo, else split[1] is index for CRUD commands.
+     *  3. Otherwise, handle single inputs with the function.
+     * @param input User input from terminal.
+     * @param d Initialised TaskList data structure
+     *
+     */
+    public void handleInput(String input, TaskList d) {
+        if (input.contains("/")) {
+            String[] split = input.split("/", 2);
+            String[] secondSplit = split[0].split(" ", 2);
+
```

Comment:
Can remove this empty line, makes code looks nicer imo

Reply:
Noted.